react-components/src/metadata/admin/EditAttributeFormList.js

```diff
@@ -175,39 +176,48 @@ class EditAttribute extends Component {
                     >
 
                         <Grid item>
-                            <Field name={`${field}.name`}
-                                   label={getMessage("attrNameLabel")}
-                                   id={build(formID, ids.ATTR_NAME)}
-                                   required={true}
-                                   margin="dense"
-                                   component={FormTextField}
+                            <FastField name={`${field}.name`}
+                                       label={getMessage("attrNameLabel")}
+                                       id={build(formID, ids.ATTR_NAME)}
+                                       required={true}
+                                       margin="dense"
+                                       component={FormTextField}
                             />
                         </Grid>
                         <Grid item>
-                            <Field name={`${field}.description`}
-                                   label={getMessage("description")}
-                                   id={build(formID, ids.ATTR_DESCRIPTION)}
-                                   component={FormTextField}
+                            <FastField name={`${field}.description`}
+                                       label={getMessage("description")}
+                                       id={build(formID, ids.ATTR_DESCRIPTION)}
+                                       component={FormTextField}
                             />
                         </Grid>
                         <Grid item>
 
-                            <Field name={`${field}.type`}
-                                   label={getMessage("attrTypeLabel")}
-                                   id={build(formID, ids.ATTR_TYPE)}
-                                   component={FormSelectField}
-                                   normalize={this.normalizeType}
-                            >
-                                {AttributeTypeMenuItems}
-                            </Field>
+                            <FastField name={`${field}.type`}
+                                       label={getMessage("attrTypeLabel")}
+                                       id={build(formID, ids.ATTR_TYPE)}
+                                       render={({ field: { onChange, ...field }, ...props }) => (
+                                           <FormSelectField
+                                               {...props}
+                                               field={field}
+                                               onChange={event => {
+                                                   this.normalizeType(event.target.value);
+                                                   onChange(event);
```

Comment:
I didn't see any post-change `normalize` support in Formik fields, so manually calling the Formik field's `onChange` function in our component's custom `onChange` function, which can do its own normalization, is one way to implement this now.